lib/fluent/plugin/filter_analyze_config.rb

```diff
@@ -122,16 +123,55 @@ module Constants
 
     include self::Constants
 
+    # Disable this warning to conform to fluentd config_param conventions.
+    # rubocop:disable Style/HashSyntax
+
     # The root configuration file of google-fluentd package.
     # This only applies to Linux.
     config_param :google_fluentd_config_path,
                  :string,
-                 default: '/etc/google-fluentd/google-fluentd.conf'
+                 :default => '/etc/google-fluentd/google-fluentd.conf'
     # Baseline configuration for comparing with local
     # customizations.
     config_param :google_fluentd_baseline_config_path,
                  :string,
-                 default: '/etc/google-fluentd/baseline/google-fluentd.conf'
+                 :default => '/etc/google-fluentd/baseline/google-fluentd.conf'
+
+    # Specify project/instance metadata.
+    #
+    # project_id, zone, and vm_id are required to have valid values, which
+    # can be obtained from the metadata service or set explicitly.
+    # Otherwise, the plugin will fail to initialize.
+    #
+    # Note that while 'project id' properly refers to the alphanumeric name
+    # of the project, the logging service will also accept the project number,
+    # so either one is acceptable in this context.
+    #
+    # Whether to attempt to obtain metadata from the local metadata service.
+    # It is safe to specify 'true' even on platforms with no metadata service.
+    config_param :use_metadata_service, :bool, :default => true
+    # A compatibility option to enable the legacy behavior of setting the AWS
+    # location to the availability zone rather than the region.
+    config_param :use_aws_availability_zone, :bool, :default => true
+    # These parameters override any values obtained from the metadata service.
+    config_param :project_id, :string, :default => nil
+    config_param :zone, :string, :default => nil
+    config_param :vm_id, :string, :default => nil
+    config_param :vm_name, :string, :default => nil
+
+    # What system to use when collecting metrics. Possible values are:
+    #   - 'prometheus', in this case default registry in the Prometheus
+    #     client library is used, without actually exposing the endpoint
+    #     to serve metrics in the Prometheus format.
+    #    - any other value will result in the absence of metrics.
+    config_param :monitoring_type, :string,
+                 :default => Monitoring::PrometheusMonitoringRegistry.name
+
+    # Override for the Google Cloud Monitoring service hostname, or
+    # `nil` to leave as the default.
+    config_param :gcm_service_address, :string, :default => nil
```

Comment:
Are these params supposed to mock the monitored resource related params in the `google_cloud` output plugin?

I wonder if we can directly take the values of these params from what users set in the `google_cloud` output plugin. That way the usage of this filter plugin remains simple even if users overwrite monitored resource. And users do not need to worry about keeping the two in sync. Since we are parsing the output plugin configuration already, we sort of have access to those params already.

What do you think?

Reply:
Discussed offline. We can just blow away these params from this plugin and take the default value.

Reply:
Thanks, pushed a new snapshot, PTAL.